File: code/sorting/insertion_sort/InsertionSort.c
/* Part of Cosmos by OpenGenus Foundation */

#include <stdio.h>

void insertion(int*, int);

int main()
{
	// n total no 
	int n;
	
	printf("how many no.\n");
	scanf("%d", &n);
	printf("enter the no. want to sort\n");
	// array name a[]
	int arr[n];
	for(int i=0;i<n;i++)
	{
		scanf("%d", &arr[i]);
	}
	// recursion call here 
	insertion(arr, n);
	return 0;
}

void insertion(int arr[], int n)
{
	int key;
	int j;
	for(int i=1;i<n;i++)
	{
		key = arr[i];
		j = i-1;
		
		while(j>=0 && arr[j]>key)
		{
			arr[j+1] = arr[j];
			j--;
		}
		arr[j+1] = key;
	}
	
	for(int i=0;i<n;i++)
	{
		printf("%d\t", arr[i]);
	}
	printf("\n");

}

File: code/sorting/insertion_sort/insertion_sort.go
package main

import (
	"fmt"
)

// Part of Cosmos by OpenGenus Foundation

func main() {
	array := []int{9, 6, 7, 3, 2, 4, 5, 1}
	insertion(array)
}

func insertion(array []int) {
	lenght := len(array)

	for i := 1; i < lenght; i++ {
		j := i

		for j > 0 && array[j-1] > array[j] {
			array[j-1], array[j] = array[j], array[j-1]

			j = j - 1
		}
	}

	fmt.Println(array)
}
